json/src/main/java/com/basistech/rosette/apimodel/jackson/recordsimilaritydeserializers/RecordSimilarityDeserializer.java

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.basistech.rosette.apimodel.jackson.recordsimilaritydeserializers;
+
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityField;
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityProperties;
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityRecords;
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityRequest;
+import com.basistech.rosette.apimodel.recordsimilarity.records.AddressRecord;
+import com.basistech.rosette.apimodel.recordsimilarity.records.DateRecord;
+import com.basistech.rosette.apimodel.recordsimilarity.records.NameRecord;
+import com.basistech.rosette.apimodel.recordsimilarity.records.Record;
+import com.basistech.rosette.apimodel.recordsimilarity.records.RecordType;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class RecordSimilarityDeserializer extends StdDeserializer<RecordSimilarityRequest> {
+
+    public RecordSimilarityDeserializer() {
+        super(RecordSimilarityRequest.class);
+    }
+
+    @Override
+    public RecordSimilarityRequest deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        try (jsonParser) {
+            final JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+            final Map<String, RecordSimilarityField> fields = node.get("fields").traverse(jsonParser.getCodec()).readValueAs(new TypeReference<Map<String, RecordSimilarityField>>() { });
+            final RecordSimilarityProperties properties = node.get("properties").traverse(jsonParser.getCodec()).readValueAs(RecordSimilarityProperties.class);
+            final RecordSimilarityRecords records = new RecordSimilarityRecords(
+                    parseRecords(node.get("records").get("left"), fields, jsonParser),
+                    parseRecords(node.get("records").get("right"), fields, jsonParser));
+            return new RecordSimilarityRequest(null, fields, properties, records);
+        }
+    }
+
+    private static List<Map<String, Record>> parseRecords(final JsonNode arrayNode,
```

Comment:
Not sure if this is the correct place, but maybe add a check for the size of the list? We want to be able to limit the number of records